astroid/nodes/scoped_nodes/scoped_nodes.py

```diff
@@ -2883,7 +2903,7 @@ def _compute_mro(self, context: InferenceContext | None = None):
                 ancestors = list(base.ancestors(context=context))
                 bases_mro.append(ancestors)
 
-        unmerged_mro = [[self], *bases_mro, inferred_bases]
+        unmerged_mro: list[list[ClassDef]] = [[self], *bases_mro, inferred_bases]
```

Comment:
.bases is currently typed as list[SuccessfulInferenceResult]
```suggestion
        unmerged_mro: list[list[SuccessfulInferenceResult]] = [[self], *bases_mro, inferred_bases]
```

Reply:
Then `clean_duplicates_mro` will rightfully complain. I think we should probably fix this function and add some assertions? But since a lot of these return `Any` now `mypy` is fine with it, for now 😄

Reply:
I was seeing a new mypy error

Reply:
Not a blocker, up to you!